papermill/iorw.py

```diff
@@ -47,7 +47,7 @@
 except ImportError:
     GCSFileSystem = missing_dependency_generator("gcsfs", "gcs")
 try:
-    from pyarrow import HadoopFileSystem
+    from pyarrow.fs import HadoopFileSystem
```

Comment:
```suggestion
    try:
      from pyarrow.fs import HadoopFileSystem
    except ImportError:
      # Attempt the older package import pattern in case we're using an old dep version.
      from pyarrow import HadoopFileSystem
```